src/pal/src/numa/numa.cpp

```diff
@@ -125,6 +106,42 @@ FreeLookupArrays()
     g_groupToCpuCount = NULL;
 }
 
+/*++
+Function:
+  AllocateLookupArrays
+
+Allocate CPU and group lookup arrays
+Return TRUE if the allocation succeeded
+--*/
+BOOL
+AllocateLookupArrays()
+{
+    g_groupAndIndexToCpu = (short*)malloc(g_groupCount * MaxCpusPerGroup * sizeof(short));
+    if (g_groupAndIndexToCpu != NULL)
+    {
+        g_cpuToAffinity = (CpuAffinity*)malloc(g_possibleCpuCount * sizeof(CpuAffinity));
+        if (g_cpuToAffinity != NULL)
+        {
+            g_groupToCpuMask = (KAFFINITY*)malloc(g_groupCount * sizeof(KAFFINITY));
+            if (g_groupToCpuMask != NULL)
+            {
+                g_groupToCpuCount = (BYTE*)malloc(g_groupCount * sizeof(BYTE));
```

Comment:
This allocation needs a null check as well, right?

Reply:
Good catch...

@janvorli Could you please change this to use the `goto done` pattern that is used in the rest of the PAL while you are on it? It is much more readable than the deeply nested if checks.

Reply:
Sigh, right.